pkg/base/capacityplanner/node_capacity.go

```diff
@@ -154,7 +154,7 @@ func (nc *nodeCapacity) selectACForVolume(vol *genV1.Volume) *accrd.AvailableCap
 	}
 
 	for _, ac := range nc.acsOrder[vol.StorageClass] {
-		if requiredSize <= nc.acs[ac].Spec.Size {
+		if requiredSize <= nc.acs[ac].Spec.Size && nc.acs[ac].Labels[v1.StorageGroupLabelKey] == vol.StorageGroup {
```

Comment:
I think for this,  you may pre-filter acs as https://github.com/dell/csi-baremetal/blob/master/pkg/base/capacityplanner/node_capacity.go#L75

Reply:
Hi Libin, it is different from the case of your disk taint feature that you can do the ACs pre-filter directly based on whether the AC has disk taint label in this initial nodeCapacity construction stage. 
In storage group scenario, it's even a common case that an ACR contains multiple volume requests to multiple storage groups resulting that we can only do the ACs filtering on different storage group for each volume request at the stage of selecting AC for the volume request, i.e. the place at selectACForVolume func you see.

Reply:
ok,  then below logic,  what if no storageGroup acs?   or every ac have a default storageGroup? 
```golang
if requiredSize <= nc.acs[ac].Spec.Size && nc.acs[ac].Labels[v1.StorageGroupLabelKey] == vol.StorageGroup
```

Reply:
No-storagegroup ACs can still work here to match the no-storagegroup volume requests (and only match no-storagegroup volume requests) here.
Those ACs without storagegroup label, i.e. no-storagegroup ACs, return "" (empty string) for nc.acs[ac].Labels[v1.StorageGroupLabelKey] while the volume requests without storage group also return "" (empty string) for vol.StorageGroup. Therefore, they can also match.

Actually, this "nil storagegroup", can be thought of as some kind of "default storagegroup" you mentioned to some extent.

At present, we have many previous tests cases without storage group can still function well on the csi images with storage group feature.

Reply:
@libzhang 
No-storagegroup ACs can still work here to match the no-storagegroup volume requests (and only match no-storagegroup volume requests) here.
Those ACs without storagegroup label, i.e. no-storagegroup ACs, return "" (empty string) for nc.acs[ac].Labels[v1.StorageGroupLabelKey] while the volume requests without storage group also return "" (empty string) for vol.StorageGroup. Therefore, they can also match.

Actually, this "nil storagegroup", can be thought of as some kind of "default storagegroup" you mentioned to some extent.

At present, all the previous tests cases without storage group can still function well on the csi images with storage group feature.

Reply:
@libzhang Has refined the code here to cover the case of AC without any labels. 
(But actually, all CSI-created ACs would contain default CSI app labels. The only very low probability of the case of AC without any labels is for the manually-created AC without labels...)